### PR TITLE
Instrumentation: Add histogram for request duration

### PR DESCRIPTION
**Pull request by bergquist**
While summaries are ok when running single instances they are not really useful when running multiple instances. 

I think we can remove the `http_request_summary` in 7.3 as histograms are better in most situations. 

This will add ~ 240 series for active instance which currently exposes ~ 550 series

Signed-off-by: bergquist <carl.bergquist@gmail.com>

### CODE REVIEW

File: pkg/middleware/request_metrics.go [hunk 1]
```diff
@@ -23,7 +24,17 @@ func init() {
 		},
 	)
 
-	prometheus.MustRegister(httpRequestsInFlight)
+	httpRequestDurationHistogram = prometheus.NewHistogramVec(
```
Comment (aknuds1) [inline pkg/middleware/request_metrics.go:27]: Is there any reason why httpRequestDurationHistogram is initialized only after declaration? Why not initialize at the same time of declaring the variable?

**Review by aknuds1 — APPROVED** [review]
Looks good AFAICT, but it looks to me as if the variable `httpRequestDurationHistogram` should be initialized at once and not in a later step.

File: pkg/middleware/request_metrics.go (hunk 1 above)
Comment (bergquist) [reply pkg/middleware/request_metrics.go:27]: I have to admit that that's the pattern I've seen in other places. But that doesn't mean it makes sense. I'll double-check.

Comment (aknuds1) [reply pkg/middleware/request_metrics.go:27]: Thanks :) I think it's safer to initialize on declaration, and this should be the pattern unless there's a reason not to.

**Review by marefr — APPROVED** [review]
Agree with Arve's feedback, but besides that looks good. Having this histogram by default is intended?

**Comment by bergquist** [comment]
Punting this to 7.3

**Comment by bergquist** [comment]
This feature requires the `http_request_histogram` feature flag to be enabled

**Review by marefr — APPROVED** [review]
LGTM

**Comment by kylebrandt** [comment]
Removed cp label because it is in beta2, and see this in the v7.3.x branch already

**Pull request merged**